### PR TITLE
Support '-b' option in 'checkout'.

### PR DESCRIPTION
**Pull request by bpeabody**

### CODE REVIEW

File: node/lib/cmd/checkout.js [hunk 1]
```diff
@@ -45,9 +45,17 @@ exports.helpText = `Check out a commit in the meta-repository and each
  */
 exports.configureParser = function (parser) {
 
+    parser.addArgument(["-b"], {
+        dest: "new branch name",
```
Comment (jhedwardyang) [inline node/lib/cmd/checkout.js]: any reason you didn't just name this "branch" (and later args.branch)?

File: node/lib/cmd/checkout.js [hunk 2]
```diff
@@ -45,9 +45,17 @@ exports.helpText = `Check out a commit in the meta-repository and each
...
+        help: `Create a new branch to check out.`,
+        nargs: 1,
+    });
+
     parser.addArgument(["committish"], {
         type: "string",
         help: "commit to check out",
+        defaultValue: null,
+        require: false,
```
Comment (jhedwardyang) [inline node/lib/cmd/checkout.js]: I think its "required", and also I believe it is invalid (doesn't work) for positional args.

Comment (jhedwardyang) [inline node/lib/cmd/checkout.js]: You're looking for `nargs: '?'`

Comment (bpeabody) [inline node/lib/cmd/checkout.js]: yes, thanks!

**Comment by jhedwardyang** [comment]
LGTM.

**Pull request merged**